packages/react-native-toast/src/components/ReactNativeToast.tsx

```diff
@@ -0,0 +1,105 @@
+import { Padding } from '@bothrs/react-native-layout'
+import {
+  FunctionComponent,
+  useState,
+  useEffect,
+  ReactNode,
+  useRef,
+} from 'react'
+import { Animated, Pressable } from 'react-native'
+import { useInterval } from 'usehooks-ts'
+
+import { useReactNativeToast } from '../hooks/useReactNativeToast'
+
+import type {
+  ReactNativeToastDirection,
+  ReactNativeToastTheme,
+} from '../types/react-native-toast.types'
+
+type ReactNativeToastProps = {
+  id: string
+  theme: ReactNativeToastTheme
+  content: ReactNode
+  direction: ReactNativeToastDirection
+  timeout?: number
+}
+
+const ReactNativeToast: FunctionComponent<ReactNativeToastProps> = ({
+  id,
+  theme,
+  content,
+  direction,
+  timeout = 3000,
+}) => {
+  const [remove, setRemove] = useState<boolean>(false)
+  const appearAnimation = useRef(new Animated.Value(0)).current
+  const [remainingTime, setRemainingTime] = useState<number>(timeout)
+  const [intervalActive, setIntervalActive] = useState<boolean>(true)
+  const { removeToast } = useReactNativeToast()
+
+  useInterval(
+    () => {
+      setRemainingTime((value) => value - 16)
+    },
+    intervalActive ? 16 : null
+  )
+
+  useEffect(() => {
+    if (remainingTime < 0 || remove) {
+      setIntervalActive(false)
+
+      if (removeToast) {
+        Animated.timing(appearAnimation, {
```

Comment:
if i'm not mistaken, The react native animated API runs on the js thread thread instead of the ui thread.

This means the animtion will only run if there is no other code being executed.

so this might result in not-so-smooth animations....


for now this is fine, but when we use this in many project we should look to use moti and/or reanimated2

Reply:
Agreed ✅ We could patch it in with `npx patch-package @bothrs/react-native-toast`